svm/src/transaction_processor.rs

```diff
@@ -364,28 +364,32 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
 
             let builtins = self.builtin_program_ids.read().unwrap().clone();
 
-            self.replenish_program_cache(
-                &account_loader,
-                &builtins,
-                &mut program_cache_for_tx_batch,
-                &mut execute_timings,
-                config.check_program_modification_slot,
-                config.limit_to_load_programs,
-                false, // increment_usage_counter
-            );
+            let ((), program_cache_us) = measure_us!({
+                self.replenish_program_cache(
+                    &account_loader,
+                    &builtins,
+                    &mut program_cache_for_tx_batch,
+                    &mut execute_timings,
+                    config.check_program_modification_slot,
+                    config.limit_to_load_programs,
+                    false, // increment_usage_counter
+                );
 
-            if program_cache_for_tx_batch.hit_max_limit {
-                return LoadAndExecuteSanitizedTransactionsOutput {
-                    error_metrics,
-                    execute_timings,
-                    processing_results: (0..sanitized_txs.len())
-                        .map(|_| Err(TransactionError::ProgramCacheHitMaxLimit))
-                        .collect(),
-                    // If we abort the batch and balance recording is enabled, no balances should be
-                    // collected. If this is a leader thread, no batch will be committed.
-                    balance_collector: None,
-                };
-            }
+                if program_cache_for_tx_batch.hit_max_limit {
```

Comment:
Don't think this `if` needs to be inside the measurement. Just makes the code harder to read.